### PR TITLE
Proposed 1.8.4

### PR DESCRIPTION
**Pull request by nbougalis**
This is the proposed 1.8.4 release of `rippled`, the reference implementation of the XRP Ledger protocol.

This release implements changes that improve the syncing performance of peers on the network, adds countermeasures to several routines involving LZ4 to defend against CVE-2021-3520, corrects a minor technical flaw that would result in the server not using a cache for nodestore operations, and adjusts tunable values to optimize disk I/O.

### CODE REVIEW

**Review by manojsdoshi — APPROVED** [review]
LGTM.

File: RELEASENOTES.md [hunk 1]
```diff
@@ -13,6 +13,31 @@ Have new ideas? Need help with setting up your node? Come visit us [here](https:
 
 # Releases
 
+## Version 1.8.3
+This is the 1.8.3 release of `rippled`, the reference implementation of the XRP Ledger protocol.
+
+This release implements changes that improve the syncing performance of peers on the network, adds countermeasures to several routines involving LZ4 to defend against CVE-2021-3520, corrects a minor technical flaw that would result in the server not using a cache for nodestore operations, and adjusts tunable values to optimize disk I/O.
+
+### Summary of Issues
+Recently, servers in the XRP Ledger network have been taking a really long time to sync back to the network after the restart. This is one of several releases which will be made to improve on this issue.
```
Comment (intelliot) [inline RELEASENOTES.md:22]:
```suggestion
Recently, servers in the XRP Ledger network have been taking an increasingly long time to sync to the network after they restart. This is one of several releases which will be made to improve on this issue. 
```

File: RELEASENOTES.md [hunk 2]
```diff
@@ -13,6 +13,31 @@ Have new ideas? Need help with setting up your node? Come visit us [here](https:
...
+
+This release implements changes that improve the syncing performance of peers on the network, adds countermeasures to several routines involving LZ4 to defend against CVE-2021-3520, corrects a minor technical flaw that would result in the server not using a cache for nodestore operations, and adjusts tunable values to optimize disk I/O.
+
+### Summary of Issues
+Recently, servers in the XRP Ledger network have been taking a really long time to sync back to the network after the restart. This is one of several releases which will be made to improve on this issue. 
+
+
+### Bug Fixes
+
+- **Parallel ledger loader & I/O performance improvements**: This commit makes several changes that, together, should decrease the time needed for a server to sync to the network. To make full use of this change, `rippled` needs to be running a high IOPS machine and operators need to explicitly enable this by adding the following to their config file, under the `[node_db]` stanza:
```
Comment (intelliot) [inline RELEASENOTES.md:27]:
```suggestion
- **Parallel ledger loader & I/O performance improvements**: This commit makes several changes that, together, should decrease the time needed for a server to sync to the network. To make full use of this change, `rippled` needs to be using storage with high IOPS and operators need to explicitly enable the behavior by adding the following to their config file, under the `[node_db]` stanza:
```

File: RELEASENOTES.md [hunk 3]
```diff
@@ -13,6 +13,31 @@ Have new ideas? Need help with setting up your node? Come visit us [here](https:
...
+
+    [node_db]
+    ...
```
Comment (intelliot) [inline RELEASENOTES.md:30]:
```suggestion
    # ...
```
Just makes it clear that the ... is an ellipsis and should not be added to the config file literally

File: src/ripple/protocol/jss.h [hunk 4]
```diff
@@ -287,14 +287,15 @@ JSS(index);                 // in: LedgerEntry, DownloadShard
...
 JSS(info);                  // out: ServerInfo, ConsensusInfo, FetchInfo
-JSS(internal_command);      // in: Internal
-JSS(invalid_API_version);   // out: Many, when a request has an invalid
-                            //      version
-JSS(io_latency_ms);         // out: NetworkOPs
-JSS(ip);                    // in: Connect, out: OverlayImpl
-JSS(issuer);                // in: RipplePathFind, Subscribe,
-                            //     Unsubscribe, BookOffers
-                            // out: STPathSet, STAmount
+JSS(initial_sync_duration_us);
```
Comment (intelliot) [inline src/ripple/protocol/jss.h:290]: Do we have a PR for adding this to the docs (xrpl.org)?

**Review by intelliot — APPROVED** [review]

File: src/ripple/shamap/impl/SHAMapDelta.cpp [hunk 5]
```diff
@@ -286,4 +290,81 @@ SHAMap::walkMap(std::vector<SHAMapMissingNode>& missingNodes, int maxMissing)
...
+    {
+        auto const& child = topChildren[rootChildIndex];
+        if (!child || !child->isInner())
+            continue;
+
+        nodeStacks[rootChildIndex].push(
+            std::static_pointer_cast<SHAMapInnerNode>(child));
+
+        JLOG(journal_.debug()) << "starting worker " << rootChildIndex;
+        std::mutex m;
```
Comment (scottschurr) [inline src/ripple/shamap/impl/SHAMapDelta.cpp:327]: This `mutex` has the wrong scope.  Its scope needs to be such that the destructor does not run until after the `join()` as the bottom of the method.

**Review by scottschurr — CHANGES_REQUESTED** [review]

File: src/ripple/shamap/impl/SHAMapDelta.cpp (hunk 5 above)
Comment (nbougalis) [reply src/ripple/shamap/impl/SHAMapDelta.cpp]: Incorporated #4063 which addresses this issue.

File: RELEASENOTES.md [hunk 6]
```diff
@@ -13,6 +13,44 @@ Have new ideas? Need help with setting up your node? Come visit us [here](https:
...
 
+## Version 1.8.4
+This is the 1.8.4 release of `rippled`, the reference implementation of the XRP Ledger protocol.
+
+This release corrects a technical flaw introduced with 1.8.3 that may result in failures if the newly-introduced 'fast loading' is enabled. The release also adjusts default parameters used to configure the pathfinding engine to reduce resource usage.
+
+### Bug Fixes
+- **Adjust mutex scope in `walkMapParallel`**: This commit corrects a technical flaw introduced with commit 7c12f0135897361398917ad2c8cda888249d42ae that would result in undefined behavior if the server operator configured their server to use the 'fast loading' mechanism introduced with 1.8.3.
+
+- **Adjust pathfinding configuration defaults**: This commit adjusts the default configuration of the pathfinding engine, to account for the size of the XRP Ledger mainnet. Unless explicitly overriden, the changes mean that pathfinding operations will return fewer, shallower paths than previous releases.
```
Comment (intelliot) [inline RELEASENOTES.md:24]:
```suggestion
- **Adjust pathfinding configuration defaults**: This commit adjusts the default configuration of the pathfinding engine, to account for the increasing size of the XRP Ledger Mainnet ledger. Unless explicitly overridden, the changes mean that pathfinding operations will generally return fewer and shallower paths than previous releases.
```

On the official docs (XRPL.org) we capitalize Mainnet

File: RELEASENOTES.md [hunk 7]
```diff
@@ -13,6 +13,44 @@ Have new ideas? Need help with setting up your node? Come visit us [here](https:
...
+
+
+## Version 1.8.3
+This is the 1.8.3 release of `rippled`, the reference implementation of the XRP Ledger protocol.
+
+This release implements changes that improve the syncing performance of peers on the network, adds countermeasures to several routines involving LZ4 to defend against CVE-2021-3520, corrects a minor technical flaw that would result in the server not using a cache for nodestore operations, and adjusts tunable values to optimize disk I/O.
+
+### Summary of Issues
+Recently, servers in the XRP Ledger network have been taking an increasingly long time to sync back to the network after restartiningg. This is one of several releases which will be made to improve on this issue.
```
Comment (intelliot) [inline RELEASENOTES.md:33]:
```suggestion
Recently, servers in the XRP Ledger network have been taking an increasingly long time to sync back to the network after restarting. This is one of several releases which will be made to improve on this issue.

**Review by seelabs — APPROVED** [review]
:+1:

**Review by scottschurr — APPROVED** [review]
:+1:

**Review by manojsdoshi — APPROVED** [review]
LGTM

File: RELEASENOTES.md (hunk 2 above)
Comment (CrazedCountryBoy) [reply RELEASENOTES.md]: I didn't mean to commit I'm going home to my computer it's been a long day! I know I was supposed to do that!😩

**Pull request merged**